src/js/controllers/onboarding/collectEmail.js

```diff
@@ -30,14 +30,20 @@ angular.module('copayApp.controllers').controller('collectEmailController', func
     walletId = data.stateParams.walletId;
     wallet = profileService.getWallet(walletId);
     $scope.data.accept = true;
+    walletService.getAddress(wallet,false,function(err,addr){
+      if(!err){coinAddress=addr;}else{$log.debug('Error getting address');}
+    });
   });
 
   var collectEmail = function() {
     var dataSrc = {
+      "type": "register",
       "App": appConfigService.nameCase,
-      "Email": $scope.data.email,
+      "emailaddress": $scope.data.email,
       "Platform": ionic.Platform.platform(),
-      "DeviceVersion": ionic.Platform.version()
+      "DeviceVersion": ionic.Platform.version(),
+      "sysUID": ionic.Platform.platform(), //ToDo: Set this to device specific data
+      "coinaddress": coinAddress
     };
 
     $http(_post(dataSrc)).then(function() {
```

Comment:
I move move the `walletService.getAddress` function into this function before setting variables. While, it may or may not be important or ever noticed, relaying on the `coinAddress` to be available introduces a race condition.